static/js/components/EmbedlyCard.js

```diff
@@ -0,0 +1,36 @@
+/* global SETTINGS: false */
+// @flow
+import React from "react"
+import isURL from "validator/lib/isURL"
+
+import { loadEmbedlyPlatform } from "../lib/embed"
+import ReactDOMServer from "react-dom/server"
+
+type Props = {
+  url: string
+}
+export default class EmbedlyCard extends React.Component<Props> {
+  componentDidMount() {
+    loadEmbedlyPlatform()
+  }
+
+  render() {
+    const { url } = this.props
+
+    if (!isURL(url, { allow_underscores: true })) {
+      return null
+    }
+
+    const html = ReactDOMServer.renderToStaticMarkup(
+      <a
+        data-card-chrome="0"
+        data-card-controls="0"
+        data-card-key={SETTINGS.embedlyKey}
+        href={url}
+        className="embedly-card"
+      />
+    )
```

Comment:
We do essentially the same thing in the ckeditor for rendering inline embeds there (rendering a div with these `data-` attributes and so on to static HTML).

maybe it would be good to pull this out into a shared function, in `lib/embed.js`?

Reply:
something with a signature like `(url: string) => html: string`

Reply:
How about `(url: string) => React.Component` for the div? Or do you think it would be better to separate the two?

Reply:
Looking at the other example, it doesn't use render with React so it does need to be `=> html: string`

Reply:
well if we had a function that returns a react component, in both of these places we're going to go ahead and immediately pass that react component to `ReactDOM.renderToStaticMarkup`, so to me it would make sense to write a function like this:

```js
const getEmbedlyCardMarkup = (url: string): string =>
  ReactDOM.renderToStaticMarkup(
    <a
      data-card-chrome="0"
      data-card-controls="0"
      data-card-key={SETTINGS.embedlyKey}
      href={url}
      className="embedly-card"
    />
  )
```

then we can call that function here and also in the `ArticleEditor`.